cypress/e2e/shared/notificationRules.test.ts

```diff
@@ -757,9 +757,426 @@ describe('NotificationRules', () => {
+                })
+              cy.get('> div')
+                .eq(1)
+                .then(elem => {
+                  expect(parseInt(elem.text().replace(/\D+/g, '')))
+                    .to.be.at.least(0)
+                    .and.to.be.at.most(Cypress.config('requestTimeout') / 1000)
+                })
+            })
+          })
+        })
+
+      // disable / enable
+      cy.get('.cf-resource-card__disabled').should('not.exist')
+      cy.getByTestID('rule-card--slide-toggle')
+        .should('have.length', Rules.length)
+        .eq(2)
+        .click()
+      cy.getByTestID('notification-error').should('not.exist')
+      cy.get('.cf-resource-card__disabled').should('have.length', 1)
+      cy.getByTestID('rule-card--slide-toggle')
+        .eq(2)
+        .click()
+      cy.getByTestID('notification-error').should('not.exist')
+      cy.get('.cf-resource-card__disabled').should('not.exist')
+
+      // verify status
+      cy.getByTestID('last-run-status--icon')
+        .eq(1)
+        .trigger('mouseover')
+      cy.getByTestID('popover--contents')
+        .should('be.visible')
+        .should(dialog => {
+          expect(dialog.text()).to.match(/^Last Run Status:.*success.*/)
+        })
+    })
+  })
+
+  describe('Cloning', () => {
+    const rule2Clone = {
+      name: 'Despair',
+      description:
+        'Nabokov.  Hermann Karlovich, a Russian of German descent and owner of a chocolate factory, meets a homeless man in the city of Prague, who he believes is his doppelgänger.',
+      label: 'Chocolate',
+    }
+
+    const newName = 'The Gift'
+    const newEvery = '5m'
+    const newOffset = '1m'
+    const newLevel = 'WARN'
+
+    beforeEach(() => {
+      cy.writeLPDataFromFile({
+        filename: 'data/wumpus01.lp',
+        offset: '20m',
+        stagger: '1m',
+      })
+      cy.get<Organization>('@org').then((org: Organization) => {
+        cy.get<Bucket>('@bucket').then((bucket: Bucket) => {
+          // get default org and bucket
+          // 2. create check
+          const queryText = `from(bucket: \"${bucket.name}\")\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\n  |> filter(fn: (r) => r[\"_measurement\"] == \"wumpus\")\n  |> filter(fn: (r) => r[\"_field\"] == \"mag\")`
+          cy.createCheck({
+            ...check,
+            orgID: org.id,
+            query: {
+              ...check.query,
+              text: queryText,
+              builderConfig: {
+                ...check.query.builderConfig,
+                buckets: [bucket.id],
+              },
+            },
+          }).then((resp: {body: any}) => {
+            cy.wrap(resp.body).as('check')
+          })
+
+          cy.createEndpoint({
+            ...(endp as NotificationEndpoint),
+            orgID: org.id,
+          }).then(resp => {
+            cy.wrap(resp.body).as('endp')
+          })
+
+          cy.get<NotificationEndpoint>('@endp').then(endp => {
+            cy.createRule({
```

Comment:
the repeated uses of cy.createCheck, cy.createEndpoint, and cy.createRule suggests some helped functions may make this easier. (plus easier to build future tests).

Not blocking for the PR. Just a suggestion.

Reply:
if you don't extract to helper functions, then please make sure to add cy.logs.

Reply:
Added a helper command to support/commands.ts